### PR TITLE
Add additional error handling and bot join/leave message filtering

### PR DESCRIPTION
**Pull request by robertlong**
This PR should filter all bot join/leave messages and provide additional information to determine the root cause of this bug. I've added sentry error logging and additional logging for Phoenix channel events.

Fixes https://github.com/MozillaReality/hubs-discord-bot/issues/98

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-* [Permissions](#permissions)
-* [Hacking on it](#hacking-on-it)
+Discord
+- [What it does](#what-it-does)
+  - [Room/channel permissions linkage](#room-channel-permissions-linkage)
+  - [Room/channel bridging](#room-channel-bridging)
+- [Running the bot](#great-i-want-to-run-this-on-my-discord-server)
+- [Permissions](#permissions)
+- [Hacking on it](#hacking-on-it)
+Slack (Alpha)
```
Comment (robin-k-wilson) [inline README.md:18]: I'd remove the slack bot stuff, since slack bot doesn't live in this repo.

File: src/reticulum.js [hunk 2]
```diff
@@ -32,12 +29,14 @@ class ReticulumChannel extends EventEmitter {
...
+      const mostRecent = p.metas[p.metas.length - 1];
+
+      if (mostRecent.context.discord) {
         return;
       }
-      const mostRecent = p.metas[p.metas.length - 1];
+
       if (curr != null && curr.metas != null && curr.metas.length > 0) {
-        // this guy was already in the lobby or room, notify iff their name changed or they moved between the lobby and room
+        // this user was already in the lobby or room, notify iff their name changed or they moved between the lobby and room
```
Comment (robin-k-wilson) [inline src/reticulum.js:39]: Nice

**Review by robin-k-wilson — COMMENTED** [review]
Double checking: on sentry + catching if the discord bot leaves and enters the room over and over, will we get a log if this behavior happens or will it silently do this?

File: src/reticulum.js [hunk 3]
```diff
@@ -47,18 +46,22 @@ class ReticulumChannel extends EventEmitter {
...
+
+      // this user was not previously present, notify for a join
       this.emit('join', Date.now(), id, mostRecent.presence, mostRecent.profile.displayName);
     });
 
     this.presence.onLeave((id, curr, p) => {
-      if (BOT_SESSION_IDS.has(id)) {
+      const mostRecent = p.metas[p.metas.length - 1];
+
+      if (mostRecent.context.discord) {
```
Comment (robin-k-wilson) [inline src/reticulum.js:57]: I wish we had typescript, will context ever be undefined?

Comment (robertlong) [reply src/reticulum.js:57]: Hmm, yeah Idk. On second glance it looks like Hubs does check for it being undefined so it probably should here as well https://github.com/mozilla/hubs/blob/master/src/react-components/presence-list.js#L111

**Comment by robertlong** [comment]
@robin-k-wilson we should get the unhandled promise errors in Sentry when this happens. We will also have more Phoenix channel logging details in journalctl. As soon as I can get our Sentry DSN, I'll see if we can setup a special alert for those errors.

**Comment by robertlong** [comment]
Deployed to prod

**Pull request merged**